app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    def check_health(self) -> None:
+        if self.health <= 0:
+            self.health = 0
+            Animal.alive.remove(self)
```

Comment:
Consider checking if `self` is in `Animal.alive` before attempting to remove it. This will prevent potential errors if `self` is not found in the list.